### PR TITLE
Progress indicator

### LINKED ISSUE #11: progress indicator
Until it can be optimized, at least the `create` command needs a progress indicator for interactive terminals with verbosity disabled.

### PR DESCRIPTION
**Pull request by lkingland**
Adds a simple progress indicator which writes a progress spinner, current steps and message.  A simple line is written if in verbose mode so as not to interfere with standard output, and nothing is written if nonverbose and a noninteractive terminal.  Fixes #11

### CODE REVIEW

File: cmd/delete.go [hunk 1]
```diff
@@ -9,9 +9,7 @@ import (
 
 func init() {
 	root.AddCommand(deleteCmd)
-
```
Comment (matejvasek) [inline cmd/delete.go:12]: It was causing a conflict, right?

File: progress/progress.go [hunk 2]
```diff
@@ -188,4 +188,5 @@ func (b *Bar) overwrite(prefix string) {
 		clear = "\033[K"
 	)
 	fmt.Fprintf(b.out, "\r%v%v%v%v/%v %v\n", up, clear, prefix, b.i, b.n, b.t)
+	// fmt.Fprintf(b.out, "\r%v%v/%v %v\033[K", prefix, b.i, b.n, b.t)
```
Comment (matejvasek) [inline progress/progress.go:191]: Why is this here?

File: progress/progress.go [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+// explicitly configured to do so, and even then not doing the continual write
+// method but the single status update in the same manner as when verbosity is
+// enabled.
+// Format:
+//   [spinner] i/n t
+type Bar struct {
+	out io.Writer
+	i   int    // Current step index
+	n   int    // Total steps
+	t   string // Current display text
```
[the remark this answers is not in the document]
Comment (lkingland) [reply progress/progress.go:33]: Sure thing, I assumed the standard symbols for total number ('n') and current index ('i') would be clear enough, but the current text ('t') is a bit cryptic.

File: progress/progress.go [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+	sync.Mutex
```
[the remark this answers is not in the document]
Comment (lkingland) [reply progress/progress.go:34]: That's strange, I receive no race conditions when building.  I am on go1.14.3 FYI

File: client.go [hunk 5]
```diff
@@ -233,30 +256,36 @@ func WithDomainSearchLimit(limit int) Option {
 // Name is derived from root if possible.
 // Root is defaulted to the current working directory.
 func (c *Client) Create(language, name, root string) (err error) {
+	c.progressListener.SetTotal(5)
```
[the remark this answers is not in the document]
Comment (lkingland) [reply client.go:259]: I agree.  This is intended to be used as a library primarily, a CLI secondarily, so relying on the os.Exit in the CLI on errors works for now, but you are right this should be a defer statement to ensure it gets run.

File: cmd/delete.go (hunk 1 above)
Comment (lkingland) [reply cmd/delete.go:12]: Conflicts were in the viper registered commands such as "-name" on multiple commands.  This was resolved by only registering the viper connetions on a command's `PreRun`.

File: progress/progress.go (hunk 2 above)
Comment (lkingland) [reply progress/progress.go:191]: This should be removed, as it was during debugging/testing.

Here's a little background about the implementations:  the initial idea was to just use `\r` to write the line, but this doesn't clear the line, so any status updates whose print statement was shorter than a previous statement resulted in dangling characters at the end from the previous print.  Adding the control character `\033[K` resolved this by explicitly clearing the line. 

This was not sufficient, however. Lines also do not include a trailing linebreak, so any status or informative text printed during execution was overwritten, the began as being appended to the status line rather than on their own line.  By using a two-line solution the error can always start after the status indication line.

File: client.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (lkingland) [reply client.go:259]: Also note that regarding the defer statement, a "defer Stop()" was used rather than deferring the Complete statement, because otherwise if the function returns mid-way through execution, the "Complete" message is printed.  This is inaccurate (if 3/5 steps are completed, the completion event did not occur so the text "5/5 Create Complete" would be a bit msleading!  The deferred Stop() simply ensures the print write is cancelled.
